### PR TITLE
feat(snakeCase): add snakeCase

### PR DESCRIPTION
**Pull request by haejunejung**
I have implemented a `snakeCase` function to address the requirements outlined in issue #120.

Following the suggestion of moving pattern to constants, I implemented caseSplitPattern in constants and test it.

image

Since there is no export environment for string, I added the export environment (package.json, ./scripts/postbuild.sh, etc) and docs for this such as intro.

### CODE REVIEW

File: benchmarks/snakeCase.bench.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { bench, describe } from 'vitest';
+import { snakeCase as snakeCaseToolkit } from 'es-toolkit';
+import { snakeCase as snakeCaseLodash } from 'lodash';
+
+describe('snakeCase', () => {
+  bench('es-toolkit/snakeCase', () => {
+    const str = 'camleCase';
```
Comment (raon0211) [inline benchmarks/snakeCase.bench.ts:7]:
```suggestion
    const str = 'camelCase';
```
Typo here :)

File: docs/ko/reference/string/snakeCase.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# snakeCase
+
+문자열을 스네이크 표기법으로 변환합니다.
+
+스네이크 케이스는 여러 단어로 구성된 식별자에서 각 단어를 소문자로 작성하고, 단어 사이를 밑줄(\_)로 연결하는 명명 규칙입니다.
```
Comment (raon0211) [inline docs/ko/reference/string/snakeCase.md:5]:
```suggestion
문자열을 스네이크 표기법으로 변환해요.

스네이크 표기법은 여러 단어로 구성된 식별자에서 각 단어를 소문자로 작성하고, 단어 사이를 밑줄(\_)로 연결하는 명명 규칙이에요. 예를 들어서, `snake_case` 처럼 작성해요.
```

File: docs/ko/reference/string/snakeCase.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+function snakeCase(str: string): string;
+```
+
+### 파라미터
+
+- `str` (`string`): 스네이크 케이스로 변환할 문자열입니다.
+
+### 반환 값
+
+(`string`) 스네이크 케이스로 변환된 문자열입니다.
```
Comment (raon0211) [inline docs/ko/reference/string/snakeCase.md:19]:
```suggestion
- `str` (`string`): 스네이크 케이스로 변환할 문자열이에요.

### 반환 값

(`string`) 스네이크 케이스로 변환된 문자열이에요.
```

File: docs/reference/string/snakeCase.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# snakeCase
+
+Converts a string to snake case.
+
+Snake case is the naming convention in which each word is written in lowercase and separated by an underscore (\_) character.
```
Comment (raon0211) [inline docs/reference/string/snakeCase.md:5]:
```suggestion
Snake case is the naming convention in which each word is written in lowercase and separated by an underscore (\_) character. For example, `snake_case`.
```

**Review by raon0211 — APPROVED** [review]
Looks great! (We might just make `getWords(...)` internally instead of just exporting a RegExp)

File: benchmarks/snakeCase.bench.ts (hunk 1 above)
Comment (haejunejung) [reply benchmarks/snakeCase.bench.ts:7]: Oh, I will fix the typo and open a pr. Thanks for the thorough code review!

Comment (haejunejung) [reply benchmarks/snakeCase.bench.ts:7]: resolved in pr #157 :)

**Pull request merged**